UnitsNet/GeneratedCode/Quantity.g.cs

```
@@ -31,18 +30,199 @@ namespace UnitsNet
+                case QuantityType.SolidAngle:
+                    return SolidAngle.From(value, SolidAngle.BaseUnit);
+                case QuantityType.SpecificEnergy:
+                    return SpecificEnergy.From(value, SpecificEnergy.BaseUnit);
+                case QuantityType.SpecificEntropy:
+                    return SpecificEntropy.From(value, SpecificEntropy.BaseUnit);
+                case QuantityType.SpecificVolume:
+                    return SpecificVolume.From(value, SpecificVolume.BaseUnit);
+                case QuantityType.SpecificWeight:
+                    return SpecificWeight.From(value, SpecificWeight.BaseUnit);
+                case QuantityType.Speed:
+                    return Speed.From(value, Speed.BaseUnit);
+                case QuantityType.Temperature:
+                    return Temperature.From(value, Temperature.BaseUnit);
+                case QuantityType.TemperatureChangeRate:
+                    return TemperatureChangeRate.From(value, TemperatureChangeRate.BaseUnit);
+                case QuantityType.TemperatureDelta:
+                    return TemperatureDelta.From(value, TemperatureDelta.BaseUnit);
+                case QuantityType.ThermalConductivity:
+                    return ThermalConductivity.From(value, ThermalConductivity.BaseUnit);
+                case QuantityType.ThermalResistance:
+                    return ThermalResistance.From(value, ThermalResistance.BaseUnit);
+                case QuantityType.Torque:
+                    return Torque.From(value, Torque.BaseUnit);
+                case QuantityType.VitaminA:
+                    return VitaminA.From(value, VitaminA.BaseUnit);
+                case QuantityType.Volume:
+                    return Volume.From(value, Volume.BaseUnit);
+                case QuantityType.VolumeFlow:
+                    return VolumeFlow.From(value, VolumeFlow.BaseUnit);
+                default:
+                    return null;
```

I think we should throw instead of return null here. I believe `ArgumentOutOfRangeException` is often used for invalid enum values, or at least R# generates that for auto-generated switch statements.

---

And maybe some test cases for the new public method `FromQuantityType`, such as also passing in an invalid enum value.

---

Done!